learners/setup.md

```diff
@@ -1,50 +1,96 @@
 
-::::::::::::::::::::::::::::::::::::::: discussion
+1. Setup workshop folder
+2. Install Python 3.11.9
+3. Setup virtual environment with required packages
+4. Download the data
 
-### Installing Python
+## 1. Setup workshop folder
 
-[Python][python] is a popular language for scientific computing, and a frequent choice
-for machine learning as well.
-To install Python, follow the [Beginner's Guide](https://wiki.python.org/moin/BeginnersGuide/Download) or head straight to the [download page](https://www.python.org/downloads/).
+Create a folder on your desktop called `dl_workshop` for storing the workshop data and required packages.
 
-Please set up your python environment at least a day in advance of the workshop.
-If you encounter problems with the installation procedure, ask your workshop organizers via e-mail for assistance so
-you are ready to go as soon as the workshop begins.
+```shell
+cd ~/Desktop
+mkdir dl_workshop
+cd dl_workshop
+pwd 
+```
 
-:::::::::::::::::::::::::::::::::::::::::::::::::::
+```output
+~/Desktop/dl_workshop
+```
 
-## Installing the required packages{#packages}
+## 2. Installing Python
 
-[Pip](https://pip.pypa.io/en/stable/) is the package management system built into Python.
-Pip should be available in your system once you installed Python successfully.
+[Python][python] is a popular language for scientific computing and a frequent choice for machine learning.
 
-Open a terminal (Mac/Linux) or Command Prompt (Windows) and run the following commands.
+Python version requirement: This workshop requires Python 3.11.9. Newer versions like 3.12 or 3.13 are not yet fully compatible with TensorFlow and may cause issues. Even Python 3.11.9 may have some edge cases, but it works well enough to be the default in Google Colab and is stable for the purposes of this workshop.
 
-1. Create a [virtual environment](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#create-and-use-virtual-environments) called `dl_workshop`:
+To install Python 3.11.9, go to the [official 3.11.9 downloads page](https://www.python.org/downloads/release/python-3119//). Choose the installer that matches your operating system (Windows, macOS, or Linux).
 
-::: spoiler
+Please set up your Python environment at least a day in advance of the workshop. If you run into issues with installation, contact the workshop organizers by email so you're ready to begin on time.
 
-### On Linux/macOs
+
+### Determine which `python` command to use for downstream setup steps
+
+Different systems and Python installations (e.g., Anaconda, Git Bash, system Python, Windows Store, etc.) may register different command names. This quick check helps identify which one points to Python 3.11.9 on your machine.
+
+Run the following in your terminal ([Git Bash recommended for Windows users](https://git-scm.com/downloads), Anaconda Prompt, or macOS/Linux shell):
 
 ```shell
-python3 -m venv dl_workshop
+python --version
+py --version
```

Comment:
I think this line has potential for confusion. Where does a `py` binary come from? Is it likely to occur when following the instructions above?
```suggestion
```